django/db/models/functions.py

```diff
@@ -1,9 +1,35 @@
 """
 Classes that represent database functions.
 """
-from django.db.models import (
-    DateTimeField, Func, IntegerField, Transform, Value,
-)
+from django.db.models import Func, Transform, Value, fields
+
+
+class Cast(Func):
+    """
+    Coerce an expression to a new field type.
+    """
+    function = 'CAST'
+    template = '%(function)s(%(expressions)s AS %(db_type)s)'
+
+    mysql_types = {
+        fields.CharField: 'char',
+        fields.IntegerField: 'signed integer',
+        fields.FloatField: 'signed',
+    }
+
+    def __init__(self, expression, output_field):
+        super(Cast, self).__init__(expression, output_field=output_field)
+
+    def as_sql(self, compiler, connection):
+        self.extra['db_type'] = self._output_field.db_type(connection)
+        return super(Cast, self).as_sql(compiler, connection)
```

Comment:
I think we want to avoid altering `self.extra` here and pass `db_type` as a kwag to `super().as_sql()`.